clients/richclient/src/main/java/org/openecard/richclient/Mounts.java

```diff
@@ -0,0 +1,194 @@
+		    String device = parts[0];
+		    String mountPath = parts[1];
+		    String fileSystem = parts[2];
+		    String mountOptions = parts[3];
+		    Map<String, String> parsedMountOptions = parseOptions(mountOptions);
+		    MountInfo mountInfo = new MountInfo(device, mountPath, fileSystem, parsedMountOptions);
+		    parsedMountInfo.add(mountInfo);
+		}
+	    }
+
+	    // convert mountInfos to mapping from mount path to MountInfo to speed up lookup of mount paths
+	    Map<String, MountInfo> map = new HashMap<>();
+	    for (MountInfo mountInfo : parsedMountInfo) {
+		map.put(mountInfo.getMountPath(), mountInfo);
+	    }
+	    instance = new Mounts(parsedMountInfo, map);
+	}
+	return instance;
+    }
+
+    /**
+     * Returns all parsed mount information in a list
+     *
+     * @return a {@code List} of all parsed {@link MountInfo MountInfos}
+     */
+    public List<MountInfo> getMountInfoList() {
+	return parsedMountInfo;
+    }
+
+    /**
+     * Returns a mapping from all mount paths to their respective {@code MountInfo}
+     *
+     * @return a mapping from all mount paths to their respective {@link MountInfo MountInfos}
+     */
+    public Map<String, MountInfo> getMountInfoMapping() {
+	return map;
+    }
+
+    /**
+     * Checks if for the provided file path the longest matching mount path has the "noexec" flag set
+     *
+     * @param filePath the path to check
+     * @return true, if the "noexec" flag for the provided file path is set; false, if the "noexec" flag is not set or
+     * if the file does not have a matching mount path in /proc/mounts (not even "/")
+     * @throws IOException if an I/O error occurs during the construction of the canonical pathname
+     * @throws NoSuchElementException if no matching mount path (not even "/") exists in /proc/mounts
+     */
+    public boolean hasNoExecFlag(String filePath) throws IOException, NoSuchElementException {
+	// get canonical file path, as the path might be a symlink
+	String canonicalFilePath = new File(filePath).getCanonicalPath();
+	// look up the mount point this path is in
+	MountInfo mountPoint = getMatchingMountPoint(canonicalFilePath);
+	// check if noexec flag is set for this mount point
+	return mountPoint.checkIfFlagIsSet("noexec");
+    }
+
+    /**
+     * Returns the longest matching mount point for the provided file path
+     *
+     * @param filePath the file path
+     * @return the longest matching {@link MountInfo} for the given path
+     * @throws IOException if an I/O error occurs during the construction of the canonical pathname
+     * @throws NoSuchElementException if no matching mount path (not even "/") exists in /proc/mounts
+     */
+    public MountInfo getMatchingMountPoint(String filePath) throws IOException, NoSuchElementException {
+	File file = new File(filePath);
+	// find the closest parent to the directory 
+	while (file != null) {
+	    String path = file.getCanonicalPath();
+	    if (map.containsKey(path)) {
```

Comment:
This is still an equals, we want to find the mount path with the longest matching prefix in the target path.

Reply:
I actually don't think that this is an equals per se. The method is provided the full target path. In the first iteration of the while-loop, it checks if that full path is already in the map, i.e. if there is an entry in /proc/mounts for the full path. If not, we repeat the loop now with the direct parent directory and check again if that path is in the map (= an entry in /proc/mounts). Rinse and repeat. As we started with the full (=the longest) path and go backwards from there, we eventually will find the mount path with the longest matching prefix for the target path.

Reply:
I see, should have looked closer.

Reply:
Maybe this means that I should've commented that better.